specifyweb/frontend/js_src/lib/components/LocalityUpdate/index.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import type { LocalizedString } from 'typesafe-i18n';
+
+import { commonText } from '../../localization/common';
+import { headerText } from '../../localization/header';
+import { localityText } from '../../localization/locality';
+import { ajax } from '../../utils/ajax';
+import type { RA } from '../../utils/types';
+import { H2 } from '../Atoms';
+import { Button } from '../Atoms/Button';
+import { formatConjunction } from '../Atoms/Internationalization';
+import { LoadingContext } from '../Core/Contexts';
+import { CsvFilePicker } from '../Molecules/CsvFilePicker';
+import { Dialog } from '../Molecules/Dialog';
+import { ProtectedAction } from '../Permissions/PermissionDenied';
+import { LocalityUpdateStatus } from './Status';
+import type { LocalityUpdateHeader } from './types';
+import {
+  localityUpdateAcceptedHeaders,
+  localityUpdateRequiredHeaders,
+} from './utils';
+
+export function LocalityUpdateFromDataSet(): JSX.Element {
+  const [headerErrors, setHeaderErrors] = React.useState({
+    missingRequiredHeaders: [] as RA<LocalityUpdateHeader>,
+    unrecognizedHeaders: [] as RA<string>,
+  });
+
+  const [headers, setHeaders] = React.useState<RA<string>>([]);
+  const [taskId, setTaskId] = React.useState<string | undefined>(undefined);
+  const [data, setData] = React.useState<RA<RA<number | string>>>([]);
+
+  const loading = React.useContext(LoadingContext);
+
+  function resetContext(): void {
+    setHeaderErrors({
+      missingRequiredHeaders: [] as RA<LocalityUpdateHeader>,
+      unrecognizedHeaders: [] as RA<string>,
+    });
+    setHeaders([]);
+  }
+
+  function handleParse(
+    columnHeaders: RA<string>,
+    data: RA<RA<number | string>>
+  ): void {
+    loading(
+      ajax('/api/localityset/parse/', {
+        method: 'POST',
+        headers: { Accept: 'application/json' },
+        body: {
+          columnHeaders,
+          data,
+          createRecordSet: false,
+          runInBackground: true,
+        },
+      }).then(({ data }) => setTaskId(data))
+    );
+  }
+
+  function handleImport(
+    columnHeaders: RA<string>,
+    data: RA<RA<number | string>>
+  ): void {
+    loading(
+      ajax('/api/localityset/import/', {
+        method: 'POST',
+        headers: { Accept: 'application/json' },
+        body: {
+          columnHeaders,
+          data,
+          createRecordSet: true,
+          runInBackground: true,
+        },
+      }).then(({ data }) => setTaskId(data))
+    );
+  }
+
+  return (
+    <ProtectedAction action="%" resource="%">
```

Comment:
uh interesting
is this to restict this to institutional admins only?
if so, could you put a short comment here?